src/useAsyncOps.js

```diff
@@ -1,37 +1,64 @@
-import { useCallback, useState } from 'react'
+import { useCallback, useState, useContext, useRef } from 'react'
 import { call as callOperation } from './operations'
+import { START, COMPLETE, FAILURE, Context, determineKey } from './Provider'
 
 export default ({
   name,
-  onStart = () => {},
-  onComplete = () => {},
-  onFailure = () => {}
+  onStart = () => { },
+  onComplete = () => { },
+  onFailure = () => { }
 }) => {
-  const [status, setStatus] = useState(false)
+  const { current: hookId } = useRef(Date.now())
+  const [runId, setRunId] = useState()
+  const [status, setStatus] = useState({ loading: false })
+  const context = useContext(Context)
+  const selector = useCallback(state => state[determineKey({ name, hookId, runId })] || {}, [name, hookId, runId])
+  const contextStatus = useContextSelector(selector)
 
   const call = useCallback(async (...args) => {
-    const data = {
-      name,
-      args
-    }
+    const runId = Date.now()
+    setRunId(runId)
+    await handleEvent(onStart, context, { type: START, hookId, runId, name, args })
+    await setStatus({ loading: true })
 
-    await onStart({ ...data })
-    setStatus({ loading: true })
     try {
       const result = await callOperation(name, ...args)
-      await onComplete({ ...data, result })
-      setStatus({ result })
+
+      await handleEvent(onComplete, context, { type: COMPLETE, hookId, runId, name, args })
+      await setStatus({ result })
+
       return result
     } catch (error) {
-      await onFailure({ ...data, error })
-      setStatus({ error })
+      await handleEvent(onFailure, context, { type: FAILURE, hookId, runId, name, args })
+      await setStatus({ error })
     }
   }, [name])
 
+  if (context) {
```

Comment:
I would like to remove all of this context logic and instead set a stub context defined in the outer closure

Reply:
I do not understand this.